### PR TITLE
NAS-119092 / 22.12.1 / Add libiscsi/pyscsi based iSCSI tests.

### PR DESCRIPTION
**Pull request by bmeagherix**
This commit adds some basic CI tests for iSCSI targets implemented using libiscsi, cython-iscsi and PYSCSI.

### CODE REVIEW

File: tests/api2/test_261_iscsi_cmd.py [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+                            assert 'Unable to connect to' in str(ve), ve
+
+                            # Finally ensure we can connect with the right CHAP creds
+                            with iscsi_scsi_connection(ip, iqn, 0, user, secret) as s:
+                                s.testunitready()
+                                data = s.inquiry().result
+                                assert data['t10_vendor_identification'].decode('utf-8').startswith("TrueNAS"), str(data)
+                                assert data['product_identification'].decode('utf-8').startswith("iSCSI Disk"), str(data)
+
+def test_06_chap():
```
Comment (yocalebo) [inline tests/api2/test_261_iscsi_cmd.py:454]: test_06_mutual_chap() maybe?

Comment (bmeagherix) [reply tests/api2/test_261_iscsi_cmd.py]: Agreed.  Thanks!

**Review by ericbsd — APPROVED** [review]
The ISCSI test LGTM

**Review by anodos325 — APPROVED** [review]
Tests look good to me as well.

**Pull request merged**